src/logic/nontrivial.lean

```diff
@@ -33,6 +33,9 @@ lemma nontrivial_iff : nontrivial α ↔ ∃ (x y : α), x ≠ y :=
 lemma exists_pair_ne (α : Type*) [nontrivial α] : ∃ (x y : α), x ≠ y :=
 nontrivial.exists_pair_ne
 
+lemma ne_or_ne_of_ne {x y z : α} (h : x ≠ y) : x ≠ z ∨ y ≠ z :=
```

Comment:
I suggest the first lemma be moved to logic.basic and renamed `ne.ne_or_ne` to enable dot notation; then it can be used to golf `decidable.exists_ne` below as @vihdzp suggested.
```
import logic.nontrivial
lemma ne.ne_or_ne {α} {a b : α} (c : α) (ab : a ≠ b) : a ≠ c ∨ b ≠ c := 
not_and_distrib.1 $ mt (and_imp.2 eq.substr) ab.symm

lemma decidable.exists_ne {α} [nontrivial α] [decidable_eq α] (x : α) : ∃ y, y ≠ x :=
let ⟨a, b, h⟩ := exists_pair_ne α in (h.ne_or_ne x).elim (λ h, ⟨a, h⟩) (λ h, ⟨b, h⟩)
```

Reply:
Oh sorry, I see you said in the other PR that you intentionally doesn't want to golf `decidable.exists_ne` because it would use axioms.

Reply:
Would you introduce `decidable.ne_or_ne_of_ne` with [decidable_eq] assumption and use [decidable.not_and_distrib](https://leanprover-community.github.io/mathlib_docs/logic/basic.html#decidable.not_and_distrib) to prove it, and then golf `decidable.exists_ne` using that?

Reply:
I personally do not mind, but I think that the point of the `decidable` namespace is to have lemmas that "control" something having to do with computability and axioms.

In particular, adding axioms to such a basic lemma seems going against the purpose that I think `decidable` has.  However, as I do not know what `decidable` really means, I may very well be wrong! 🙃

Reply:
> Would you introduce `decidable.ne_or_ne_of_ne` with [decidable_eq] assumption and use [decidable.not_and_distrib](https://leanprover-community.github.io/mathlib_docs/logic/basic.html#decidable.not_and_distrib) to prove it, and then golf `decidable.exists_ne` using that?

Honestly, I do not know: I am a mathematician, for me decidable is a synonym of "someone knows a proof of this fact"!

Reply:
I moved the lemma to `logic.basic`, but will wait introducing decidable stuff unless clearly instructed to do so!  :smile: